package systemd unit files

Olf0 opened the pull request:
- enable systemd in configure
- package fstrim.timer
- package uuidd.service

as per https://gitlab.com/nephros/sailfishos-util-linux/-/compare/master...50a6d02b?from_project_id=28856859
by @nephros

Background information: https://forum.sailfishos.org/t/feature-suggestion-install-and-use-systemds-fstrim-timer/7634
(originally at https://together.jolla.com/question/219907/please-install-and-use-systemds-fstrimtimer/)

Olf0 committed:
package systemd unit files

- enable systemd in configure
- package fstrim.timer
- package uuidd.service

as per https://gitlab.com/nephros/sailfishos-util-linux/-/compare/master...50a6d02b?from_project_id=28856859

Background information: https://forum.sailfishos.org/t/feature-suggestion-install-and-use-systemds-fstrim-timer/7634

Olf0 commented:
P.S.: For reference and testing also see https://build.sailfishos.org/package/show/home:nephros/util-linux
**but** the latest build there includes an additional commit (to enable easy testing): https://gitlab.com/nephros/sailfishos-util-linux/-/compare/master...c9a77120?from_project_id=28856859
>Tone down dependencies for the systemd units
...so they can be installed independently of main package version.
Intended for testing.
As newer versions in upstream use different options to the utility this should not be done in any release version of this package, as it may break in a future version.

okodron commented on rpm/util-linux.spec:
```diff
@@ -198,6 +198,26 @@ Obsoletes: %{name}-docs
+Requires:  %{name} = %{version}-%{release}
+Requires:  uuidd
+Requires(post): systemd
+BuildRequires:  systemd-devel
```
`BuildRequires` should be add once in the beginning of the spec as it's not sub-package specific.
And it's better to use pkgconfig:
```
BuildRequires: pkgconfig(systemd)
```

Thaodan replied:
Requires(post) is fine but I agree use pkgconfig please for BuildRequires.

Olf0 replied:
O.K., done (both aspects of @okodron's request).
Can you please mark this as "viewed".

Olf0 committed:
Use a single BuildRequires: pkgconfig(systemd)

... at the start of this spec file, instead of two sub-package specific statements, as requested by @okodron.
Also use BuildRequires: pkgconfig(systemd) instead of BuildRequires: systemd-devel

Olf0 committed:
Remove accidentially introduced blank line

okodron commented on rpm/util-linux.spec:
```diff
@@ -198,6 +199,24 @@ Obsoletes: %{name}-docs
 
+%package systemd-fstrim
+Summary:   Systemd unit files for %{name}
+Group:     System
```
Please, drop `Group:` as we don't use it any more.

Olf0 replied:
Please @okodron, do implement cosmetic changes, if you have any additional ones:
- This PR is explicitly marked as "maintainers may alter it".
- As a community contributor, I am doing this in my spare time, you in your paid time.
- I already contributed the suggestion (2 years ago), plus now an implementation (by @nephros), which is tested (by him and me).
- As a community contributor, Jolla's non-public style-guides and its changes ("we don't use this anymore") are unknown to me.

Hence feel free to perform further adaptions, as long as the intended functionality is retained.
Note that I am sure willing to continue discussing technical aspects of this PR.

Olf0 replied:
@okodron, when thoroughly reviewing your request in the spec file, it becomes obvious that the principal package and *every* sub-package carries a `Group:` statement.  Hence defining the apropiate Groups for the two newly added sub-packages is aligned to the rest of the spec file style.  If you want to change the current style, you may do that, but that would definitely expand the scope of this PR, hence I suggest to perform this change in a separate PR, because it is unrelated to the content and intention of this PR, plus such a style change should be properly and visibly (hence separately) documented.

okodron replied:
Ok, it can be done in a separate MR cause it's not strictly related to these changes.

okodron commented on rpm/util-linux.spec:
```diff
@@ -198,6 +199,24 @@ Obsoletes: %{name}-docs
+Requires(post): systemd
+
+%description systemd-fstrim
+Systemd unit files for fstrim
```
Description should end with full stop to make rpmlint happy:
```suggestion
Systemd unit files for fstrim.
```

Olf0 committed:
Add full stop

... as requested by @okodron:
"Description should end with full stop to make rpmlint happy:"

Co-authored-by: Andrey Okoshkin <okoandrey@gmail.com>

okodron commented on rpm/util-linux.spec:
```diff
@@ -198,6 +199,24 @@ Obsoletes: %{name}-docs
+Requires(post): systemd
+
+%description systemd-uuidd
+Systemd unit files for uuidd
```
The same issue with rpmlint here.

Olf0 replied:
O.K., [done](https://github.com/sailfishos/util-linux/pull/2/commits/8f0e9bfe3f4f802c07394c892b3006dfc04afb12).

Olf0 replied:
Nevertheless, still [please do implement cosmetic changes, if you have any additional ones: This PR is explicitly marked as "maintainers may alter it" etc. [...]](https://github.com/sailfishos/util-linux/pull/2#discussion_r699330306)

Thaodan replied:
@Olf0 Its normal that if you want to contribute that you have to follow the guidelines of the platform.
In our case see here the specific for packaging: https://sailfishos.org/wiki/Software_Packaging

Olf0 committed:
Add full stop

... as requested by @okodron: "Description should end with full stop to make rpmlint happy."

Thaodan reviewed (changes requested):
Please smash your fixup commit together with the original commit and don't add groups.
See the links below for more information:
https://sailfishos.org/wiki/Software_Packaging
https://sailfishos.org/wiki/Collaborative_Development


>As a community contributor, Jolla's non-public style-guides and its changes ("we don't use this >anymore") are unknown to me.

Jolla's guidelines on this are public as shown above ;)
Please look into them and come back to me if something is missing on our side.

Olf0 committed:
Delete the two added Group: statements

... for the newly added sub-packages (in contrast to all extant sub.packages), as requested by @Thaodan.

Olf0 commented:
> Please smash your fixup commit together with the original commit and don't add groups.

Oh my ...; well, see PR #3.

Thaodan commented:
> > Please smash your fixup commit together with the original commit and don't add groups.
> 
> Oh my ...; well, see PR #3.

For the future please look into this about rebasing and squashing commits:
https://git-scm.com/docs/git-rebase#_interactive_mode
